lib/eslint.config_partial.mjs

```diff
@@ -13,6 +13,15 @@ const noRestrictedSyntax = [
     selector: "CallExpression[callee.object.name='assert']:not([callee.property.name='ok']):not([callee.property.name='fail']):not([callee.property.name='ifError'])",
     message: 'Only use simple assertions',
   },
+  // Forbids usages of `btoa` like:
+  // ```
+  // const { btoa } = internalBinding('buffer');
+  // btoa('...');
+  // ```
+  {
```

Comment:
nit: I would put the comment inside the block (in case the block gets moved around, it's less likely to leave the comment behind), and I would specifically call out `no-restricted-globals` since that's what I was confused about.
```suggestion
  {
    // Forbids usages of `btoa` that are not caught by no-restricted-globals, like:
    // ```
    // const { btoa } = internalBinding('buffer');
    // btoa('...');
    // ```
```